spec/test.spec.js

```diff
@@ -514,102 +551,120 @@ describe('S3Adapter tests', () => {
       options = {
         bucketPrefix: 'test/',
       };
+      s3ClientMock = jasmine.createSpyObj('S3Client', ['send']);
+      s3ClientMock.send.and.returnValue(Promise.resolve());
+    });
+
+    it('should save a file with right command', async () => {
+      const s3 = getMockS3Adapter(options);
+      s3._s3Client = s3ClientMock;
+
+      await s3.createFile('file.txt', 'hello world', 'text/utf8', {});
+      expect(s3ClientMock.send.calls.all().length).toBe(2);
+      expect(s3ClientMock.send).toHaveBeenCalledWith(jasmine.any(CreateBucketCommand));
+      expect(s3ClientMock.send).toHaveBeenCalledWith(jasmine.any(PutObjectCommand));
     });
 
     it('should save a file with metadata added', async () => {
       const s3 = getMockS3Adapter(options);
-      s3._s3Client.upload = (params, callback) => {
-        const { Metadata } = params;
-        expect(Metadata).toEqual({ foo: 'bar' });
-        const data = {
-          Body: Buffer.from('hello world', 'utf8'),
-        };
-        callback(null, data);
-      };
-      const fileName = 'randomFileName.txt';
+      s3._s3Client = s3ClientMock;
       const metadata = { foo: 'bar' };
-      await s3.createFile(fileName, 'hello world', 'text/utf8', { metadata });
+
+      await s3.createFile('file.txt', 'hello world', 'text/utf8', { metadata });
+      const commandArg = s3ClientMock.send.calls.mostRecent().args[0];
+      expect(commandArg.input.Metadata).toEqual({ foo: 'bar' });
     });
 
     it('should save a file with tags added', async () => {
       const s3 = getMockS3Adapter(options);
-      s3._s3Client.upload = (params, callback) => {
-        const { Tagging } = params;
-        expect(Tagging).toEqual('foo=bar&baz=bin');
-        const data = {
-          Body: Buffer.from('hello world', 'utf8'),
-        };
-        callback(null, data);
-      };
-      const fileName = 'randomFileName.txt';
+      s3._s3Client = s3ClientMock;
       const tags = { foo: 'bar', baz: 'bin' };
-      await s3.createFile(fileName, 'hello world', 'text/utf8', { tags });
+
+      await s3.createFile('file.txt', 'hello world', 'text/utf8', { tags });
+      const commandArg = s3ClientMock.send.calls.mostRecent().args[0];
+      expect(commandArg.input.Tagging).toBe('foo=bar&baz=bin');
     });
 
     it('should save a file with proper ACL with direct access', async () => {
-      // Create adapter
       options.directAccess = true;
       const s3 = getMockS3Adapter(options);
-      spyOn(s3._s3Client, 'upload').and.callThrough();
-      // Save file
+      s3._s3Client = s3ClientMock;
+
       await s3.createFile('file.txt', 'hello world', 'text/utf8', {});
-      // Validate
-      const calls = s3._s3Client.upload.calls.all();
-      expect(calls.length).toBe(1);
-      calls.forEach((call) => {
-        expect(call.args[0].ACL).toBe('public-read');
-      });
+      const commandArg = s3ClientMock.send.calls.mostRecent().args[0];
+      expect(commandArg.input.ACL).toBe('public-read');
```

Comment:
This and other tests seems to not ensure anymore that there was only 1 call. Is that because the adapter has changed, or did that just get lost in the migration?

Reply:
Unfortunately, I don’t remember why that got lost 😅, and I’m not sure why we need this test again. But if you still think it's necessary, I can add it back in!

Reply:
I haven't looked into the purpose of these lines, but if they don't cause any issues adding them back, then I'd just add them back.

Reply:
I added these checks and command instance checks as well